subsys/net/ip/ipv6.c

```diff
@@ -4251,15 +4261,14 @@ enum net_verdict net_ipv6_process_pkt(struct net_pkt *pkt)
 
 		if (is_upper_layer_protocol_header(next)) {
 			NET_DBG("IPv6 next header %d", next);
-			net_pkt_set_ipv6_ext_len(pkt, offset -
-						 sizeof(struct net_ipv6_hdr));
+			net_pkt_set_ipv6_ext_len(pkt, length);
```

Comment:
As there can be multiple extension headers, we need to accumulate the total length here.